### ISSUE TITLE
Translation example weird loss behavior

### ISSUE THREAD

**Issue by Guitaricet** [pytorch/fairseq#1775]
## 🐛 Bug

Training loss changes abruptly after validation.

<img width="1258" alt="Screen Shot 2020-03-04 at 5 43 03 PM" src="https://user-images.githubusercontent.com/2821124/75929972-a3116c00-5e3f-11ea-8f1d-a07ac9e11d77.png">

(red dot is validation loss after the first epoch)

### To Reproduce

```bash
bash prepare-wmt14en2de.sh
cd ../..

TEXT=examples/translation/wmt17_en_de
fairseq-preprocess \
    --source-lang en --target-lang de \
    --trainpref $TEXT/train --validpref $TEXT/valid --testpref $TEXT/test \
    --destdir data-bin/wmt17_en_de \
    --workers 10

MODEL_NAME=transformer_en_de_Mar04_v2
mkdir -p checkpoints/$MODEL_NAME

CUDA_VISIBLE_DEVICES=0,1,2 fairseq-train\
     data-bin/wmt17_en_de\
     --arch transformer --share-decoder-input-output-embed\
     --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0\
     --lr 5e-4 --lr-scheduler inverse_sqrt --warmup-updates 4000\
     --dropout 0.3 --weight-decay 0.0001\
     --criterion label_smoothed_cross_entropy --label-smoothing 0.1\
     --max-tokens 3200 \
     --max-update 300000 \
     --tensorboard-logdir logs/$MODEL_NAME\
     --save-dir checkpoints/$MODEL_NAME\
     --eval-bleu \
     --eval-bleu-args '{"beam": 5, "max_len_a": 1.2, "max_len_b": 10}' \
     --eval-bleu-detok moses \
     --eval-bleu-remove-bpe \
     --best-checkpoint-metric bleu --maximize-best-checkpoint-metric \
```

#### Code sample
Did not change fairseq code

### Expected behavior

Loss does not change after validation.

### Environment

 - fairseq Version (e.g., 1.0 or master): master (c699eb054e2782e2ae4e1d85d1fff5a724dc5c0e)
 - PyTorch Version (e.g., 1.0): 1.4
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): `pip install --editable .`
 - Python version: 3.7
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: 3x GTX 1080
 - Any other relevant information: apex is installed (ignoring CUDA version mismatch) 

### Additional context

Training loss of the first batch after validation is almost exactly validation loss (5.029 vs 4.952).
I would suggest that fairseq performs exponential smoothing while logging the loss value validation loss point leaks to training loss points.

**Comment by Guitaricet** [comment]
This jump in loss function happends every epoch.

<img width="1266" alt="Screen Shot 2020-03-04 at 10 36 00 PM" src="https://user-images.githubusercontent.com/2821124/75945547-8fc6c680-5e68-11ea-84c1-be81b12088c9.png">

**Comment by myleott** [comment]
Historically fairseq reported the running average over the epoch. We just recently changed the behavior so that it averages over the log interval instead: https://github.com/pytorch/fairseq/commit/244835d811c2c66b1de2c5e86532bac41b154c1a

This was merged right after the commit you’re using :)

**Comment by Guitaricet** [comment]
Thank you, I guess I'm lucky in a way (=
